device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -146,7 +142,7 @@
     void open() throws IOException
     {
         /* Codes_SRS_DEVICE_IO_21_007: [If the client is already open, the open shall do nothing.] */
-        if (this.state == IotHubClientState.OPEN)
+        if (this.state == IotHubConnectionStatus.CONNECTED || this.state == IotHubConnectionStatus.DISCONNECTED_RETRYING)
```

Comment:
how are these managed when calls are made in parallel - i.e. I call open() over one thread, and then close() over another thread, at the same time.

Reply:
I can add a state lock for that, sure. Right now, we just hope customers don't do that